### PR TITLE
feat: Backoff decorator for jobs scheduling

### PR DESCRIPTION
**Pull request by jgolebiowski**
Simple backoff decorator which can be used for scheduling sagemaker jobs. Once a function is decorated with it, it will run normally but wait on a specific error (e.g. ResourceShareLimitExceededException) and retry after some time. This can be used to leave a scheduling loop running in the background and have it allocate new jobs when instances become available.

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/backend/scheduling.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (mseeger) [inline syne_tune/backend/scheduling.py:1]: I think a better place for this would be `syne_tune/remote`, no?

**Review by mseeger — COMMENTED** [review]
Once this is in, I'd like to integrate it with benchmarking (I can do that)

File: syne_tune/backend/scheduling.py (hunk 1 above)
Comment (jgolebiowski) [reply syne_tune/backend/scheduling.py]: Good idea, moved.

**Comment by jgolebiowski** [comment]
> Once this is in, I'd like to integrate it with benchmarking (I can do that)

That would be great, if you can do this then its perfect for me. Otherwise we can have a quick chat since I am not yet certain where is the best place to integrate.

**Pull request merged**